src/asn_combinators.ml

```diff
@@ -139,6 +139,40 @@ let int =
   in
   map f g integer
 
+let unsigned_integer =
+  let f str =
+    let l = String.length str in
+    if l > 0 then
+      let fst = string_get_uint8 str 0 in
+      if fst > 0x7F then
+        parse_error "unsigned integer < 0"
+      else if fst = 0x00 then
+        String.sub str 1 (l - 1)
+      else
+        str
+    else
+      str
+  and g str =
+    let l = String.length str in
+    let rec strip0 off =
+      if l - off >= 2 &&
+         string_get_uint8 str off = 0x00 &&
+         string_get_uint8 str (off + 1) < 0x80
+      then
+        strip0 (off + 1)
+      else if off = 0 then
+        str
+      else
+        String.sub str off (l - off)
+    in
+    let str' = strip0 0 in
+    if String.length str' = 0 || string_get_uint8 str' 0 > 0x7F then
+      "\x00" ^ str'
+    else
+      str'
+  in
+  map f g integer
```

Comment:
We *could* pass a `~random` in order to avoid the `try .. with` in the tests which makes the test somewhat weaker. However, it seems to me it is a bit annoying to write such a generator so I'm fine with it as is.

Reply:
Oh, indeed we can do this as well, similar to how the time is constructed. I like that approach better - we can as well use it for int vs integer...

Reply:
done in [ec9943a](https://github.com/mirleft/ocaml-asn1-combinators/pull/44/commits/ec9943afaef3d7f84fe13887587c077660aff77f)